recon/src/main/java/org/hps/recon/utils/TrackClusterMatcherMinDistance.java

```diff
@@ -0,0 +1,753 @@
+         * Cluster.
+         * This matcher works for both GBL and KF Tracks.
+         * Returns Map of Tracks matched do unique Clusters
+         */
+
+        //Initialize clusterToTrack map, used for applying cluster corrections
+        this.clusterToTrack = new HashMap<Cluster, Track>();
+
+        // Map stores all Track Cluster pairs in Event.
+        HashMap<Track, Cluster> alltrackClusterPairs = new HashMap<Track, Cluster>();
+
+        //GBL Tracks require rotation tables
+        if(this.trackCollectionName.contains("GBLTracks")){
+            hitToRotated = TrackUtils.getHitToRotatedTable(event);
+            hitToStrips = TrackUtils.getHitToStripsTable(event);
+        }
+
+        //For KF
+        //Relational Table is used to get KF track time from
+        //KFTrackDataRelations
+        if(this.trackCollectionName.contains("KalmanFullTracks")){
+            List<TrackData> TrackData;
+            List<LCRelation> trackRelations;
+            TrackData trackdata;
+            if (this.trackCollectionName.contains("KalmanFullTracks")) {
+                TrackData = event.get(TrackData.class, "KFTrackData");
+                trackRelations = event.get(LCRelation.class, "KFTrackDataRelations");
+                for (LCRelation relation : trackRelations) {
+                    if (relation != null && relation.getTo() != null){
+                        trackToData.add(relation.getFrom(), relation.getTo());
+                    }
+                }
+            }
+        }
+
+        if(enablePlots){
+            plotClusterValues(clusters,cuts.getTrackClusterTimeOffset());
+        }
+
+        //For all Track lists in trackCollections, match Tracks to Clusters
+        for (List<Track> tracks : trackCollections) {
+
+            if(tracks == null || tracks.isEmpty() || clusters == null || clusters.isEmpty()){
+                continue;
+            }
+
+            //Map relating Tracks to potential Cluster matches, and their
+            //position residuals
+            Map<Track, Map<Cluster, Double>> trackClusterResidualsMap = new HashMap<Track, Map<Cluster, Double>>(); 
+
+            for(Track track : tracks) {
+
+                 // Derive the charge of the particle from the track.
+                int charge = (int) Math.signum(track.getTrackStates().get(0).getOmega())*flipSign;
+
+                //Track time includes a hard-coded offset
+                double trackt = getTrackTime(track);
+                double tracktOffset = 3.5; //Track time distribution is centered on -4ns, added offset to center ~0
+
+                //Track positions depend on GBL or KF
+                List<Double> trackPos = this.getTrackPositionAtEcal(track);
+                double trackx = trackPos.get(0);
+                double tracky = trackPos.get(1);
+                double trackz = trackPos.get(2);
+
+                //dxoffset to be determined using truth information 03.19.21
```

Comment:
Same

Reply:
These are the offsets that were originally hard-coded that you asked me to just remove for now. Do you want these to be configurable from the steering-file for the time being?